docs/user/design_lifecycle.md

```diff
@@ -0,0 +1,67 @@
+# Design LifeCycle
+
+<!-- TODO: Add the screenshoots -->
+
+According to a design-oriented approach, the Design Builder App provides not only with the capacity to create and update data in Nautobot but also with a complete lifecycle management of each deployment: update, versioning (in the future), and decommissioning.
+
+<!-- TODO: without an identifier: IDENTIFIER_KEYS = ["!create_or_update", "!create", "!update", "!get"],
+the update features are not working as expected.
+I would propose to use explicit action tags even for create: "!create:name"
+ -->
+
+All the Design Builder UI navigation menus are under the Design Builder tab.
+
+## `Design`
+
+A `Design` is a one to one mapping with a Nautobot `Job`, enriched with some data from the Design Builder `DesignJob` definition. In concrete, it stores:
+
+- A `Job` reference.
+- A `version` string from the `DesignJob`.
+- A `description` string from the `DesignJob`.
+- A `docs` string from the `DesignJob`.
+
+<!-- TODO: Add the screenshoot of the table of Design -->
+
+From the `Design`, the user can manage the associated `Job`, and trigger its execution to deploy it creating a `DesignInstance` or Design Deployment
+
+## Design Deployment or `DesignInstance`
+
+Once a design is "deployed" in Nautobot, a Design Deployment (or `DesignInstance`) is created with the report of the changes implemented (i.e. `Journals`), and with actions to update or decommissioning it (see next subsections).
+
+The `DesignInstance` stores:
+
+- The `name` of the deployment, within the context of the `Design`.
+- The `Design` reference.
+- The `version` from the `Design` when it was deployed or updated.
+- When it was initially deployed or last updated.
+- The `status` of the design, and the `live_state` or operational status to signal its state in the actual network.
+
+<!-- TODO: Add the screenshoot of the table of DesignInstance -->
+
+### Design Deployment Update
+
+This feature allows to re run a design instance with different input data to update the implemented design with the new changes: additions and removals.
+
+It leverages a complete tracking of previous design implementation and a reduce function for the new design to understand the changes to be implemented and the objects to be decommissioned (leveraging the previous decommissioning feature for only a specific object).
```

Comment:
We need to really consider the term `reduce`. I think from a mathematical/computer science perspective the operation is more like a `map` than a `reduce`.

Reply:
yes, taking it into account